### PR TITLE
Abnormal Tier Usage: Two Streams Added

### PR DESCRIPTION
**Pull request by upul**

### CODE REVIEW

File: features/org.wso2.carbon.analytics.apim.feature/src/main/capp/Eventstreams/org_wso2_analytics_apim_abnormalTierUsageAlert_1.0.0/org.wso2.analytics.apim.alert.abnormalTierUsageAlert_1.0.0.json [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  "correlationData": [],
+  "payloadData": [
+    {
+      "name": "userId",
+      "type": "STRING"
+    },
+    {
+      "name": "api",
+      "type": "STRING"
+    }
```
Comment (nirmal070125) [inline features/org.wso2.carbon.analytics.apim.feature/src/main/capp/Eventstreams/org_wso2_analytics_apim_abnormalTierUsageAlert_1.0.0/org.wso2.analytics.apim.alert.abnormalTierUsageAlert_1.0.0.json]: All alert streams should have a proper alert message field.

File: features/org.wso2.carbon.analytics.apim.feature/src/main/capp/Sparkscripts/org_wso2_analytics_apim_abnormal_tier_usage_1.0.0/org_wso2_analytics_apim_abnormal_tier_usage.xml [hunk 2]
```diff
@@ -19,12 +21,38 @@
...
       USING CarbonAnalytics
-      OPTIONS (tableName "ORG_WSO2_API__ABN_REQ_ALERT_TBL",
+      OPTIONS (tableName "ORG_WSO2_API_ALERT_STORE_TBL",
          schema "userId STRING, api STRING"
       );
 
+      CREATE TEMPORARY TABLE ORG_WSO2_API_ABN_REQ_ALERT
+      USING org.wso2.carbon.analytics.spark.event.EventStreamProvider
+      OPTIONS (receiverURL "tcp://10.100.5.119:7612",
+               authURL "ssl://10.100.5.119:7712",
```
Comment (nirmal070125) [inline features/org.wso2.carbon.analytics.apim.feature/src/main/capp/Sparkscripts/org_wso2_analytics_apim_abnormal_tier_usage_1.0.0/org_wso2_analytics_apim_abnormal_tier_usage.xml]: Please use localhost

File: features/org.wso2.carbon.analytics.apim.feature/src/main/capp/Sparkscripts/org_wso2_analytics_apim_abnormal_tier_usage_1.0.0/org_wso2_analytics_apim_abnormal_tier_usage.xml [hunk 3]
```diff
@@ -19,12 +21,38 @@
...
+               version "1.0.0",
+               description "This is the CEP stream for Abnormal Tier Usage",
+               nickName "abnormalTierUsageAlert",
+               payload "userId STRING, api STRING"
+      );
+
+      CREATE TEMPORARY TABLE ORG_WSO2_API_ALL_ALERT
+      USING org.wso2.carbon.analytics.spark.event.EventStreamProvider
+      OPTIONS (receiverURL "tcp://10.100.5.119:7612",
+               authURL "ssl://10.100.5.119:7712",
```
Comment (nirmal070125) [inline features/org.wso2.carbon.analytics.apim.feature/src/main/capp/Sparkscripts/org_wso2_analytics_apim_abnormal_tier_usage_1.0.0/org_wso2_analytics_apim_abnormal_tier_usage.xml]: Please use localhost

File: features/org.wso2.carbon.analytics.apim.feature/src/main/capp/Sparkscripts/org_wso2_analytics_apim_abnormal_tier_usage_1.0.0/org_wso2_analytics_apim_abnormal_tier_usage.xml [hunk 4]
```diff
@@ -19,12 +21,38 @@
...
          schema "userId STRING, api STRING"
       );
 
+      CREATE TEMPORARY TABLE ORG_WSO2_API_ABN_REQ_ALERT
+      USING org.wso2.carbon.analytics.spark.event.EventStreamProvider
+      OPTIONS (receiverURL "tcp://10.100.5.119:7612",
+               authURL "ssl://10.100.5.119:7712",
+               username "admin",
+               password "admin",
+               streamName "org.wso2.analytics.apim.alert.abnormalTierUsageAlert",
```
Comment (nirmal070125) [inline features/org.wso2.carbon.analytics.apim.feature/src/main/capp/Sparkscripts/org_wso2_analytics_apim_abnormal_tier_usage_1.0.0/org_wso2_analytics_apim_abnormal_tier_usage.xml]: We have to check whether we should use plain text passwords here. @lafernando @mohanvive any thoughts?

File: features/org.wso2.carbon.analytics.apim.feature/src/main/capp/Sparkscripts/org_wso2_analytics_apim_abnormal_tier_usage_1.0.0/org_wso2_analytics_apim_abnormal_tier_usage.xml [hunk 5]
```diff
@@ -40,11 +68,19 @@
...
      WHERE S.userId = D.userId AND S.api = D.api AND S.request_count >=  D.requestsPerDayPercentile
      GROUP BY S.userId, S.api
      HAVING COUNT(S.api) >= 5;
+
+     INSERT OVERWRITE TABLE ORG_WSO2_API_ABN_REQ_ALERT
+     SELECT userId, api
+     FROM ORG_WSO2_API_ALERT_STORE;
+
+     INSERT OVERWRITE TABLE ORG_WSO2_API_ALL_ALERT
+     SELECT "Abnormal Tier Usage", concat(concat(userId, ':'), api)
```
Comment (nirmal070125) [inline features/org.wso2.carbon.analytics.apim.feature/src/main/capp/Sparkscripts/org_wso2_analytics_apim_abnormal_tier_usage_1.0.0/org_wso2_analytics_apim_abnormal_tier_usage.xml]: Please construct a meaningful message with all the information that are relevant and available.

File: features/org.wso2.carbon.analytics.apim.feature/src/main/capp/Sparkscripts/org_wso2_analytics_apim_abnormal_tier_usage_1.0.0/org_wso2_analytics_apim_abnormal_tier_usage.xml (hunk 4 above)
Comment (mohanvive) [inline features/org.wso2.carbon.analytics.apim.feature/src/main/capp/Sparkscripts/org_wso2_analytics_apim_abnormal_tier_usage_1.0.0/org_wso2_analytics_apim_abnormal_tier_usage.xml]: Not sure, whether spark script supports to have encrypted passwords.. @lafernando any thoughts ? But, In CEP artifacts (like event receivers, event publishers), passwords are get encrypted by default..

Comment (lafernando) [inline features/org.wso2.carbon.analytics.apim.feature/src/main/capp/Sparkscripts/org_wso2_analytics_apim_abnormal_tier_usage_1.0.0/org_wso2_analytics_apim_abnormal_tier_usage.xml]: @dunithd @nirandaperera do we have password encryption support there? .. If not, it is something we should get implemented.

**Pull request merged**